backend/substrapp/ledger_utils.py

```diff
@@ -157,6 +161,44 @@ def _wrapper(*args, **kwargs):
     return _retry
 
 
+def to_snake_case(camel_str):
+    name = CAMEL_TO_SNAKE_PATTERN.sub(r'\1_\2', camel_str)
+    name = CAMEL_TO_SNAKE_PATTERN_2.sub(r'\1_\2', name).lower()
+    return name.replace('_i_d', '_id')
+
+
+def _replace_dict_keys(d, converter):
+    """Replace fields in a dict and return updated dict (recursive).
+
+    Apply converter to each dict field.
+    """
+    assert isinstance(d, dict)
+
+    new_d = {}
+    for key, value in d.items():
+        if isinstance(value, dict):
+            value = _replace_dict_keys(value, converter)
+        elif isinstance(value, list):
+            if all([isinstance(v, dict) for v in value]):
+                value = [_replace_dict_keys(v, converter) for v in value]
+
+        new_d[converter(key)] = value
+    return new_d
```

Comment:
there is a "metadata" property for all assets that the user can fill with a dict. The keys of this dict shouldn't be converted to snake_case since they are user input.